src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/fallthrough/InputFallThrough4.java

```diff
@@ -0,0 +1,31 @@
+/*
+FallThrough
+checkLastCaseGroup = (default)false
+reliefPattern = (default)falls?[ -]?thr(u|ough)
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.coding.fallthrough;
+
+public class InputFallThrough4
+{
+    void foo(int i){
+        switch(i) {
+            case 1:
+                 int fallthru = 2; // fallthru
+            case 2:// violation 'Fall through from previous branch of the switch statement.'
+                 int fallthrough = 2; //            fall thru
+            case 3: // violation 'Fall through from previous branch of the switch statement.'
```

Comment:
Why are these violations? Shouldn't they be ok since we have fallthru comment?

Reply:
@nrmancuso The violations are also shown in the CLI o/p;

```hp@DESKTOP-OPBFIIB MINGW64 ~/Desktop
$ cat config1.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="FallThrough">
            <property name="checkLastCaseGroup" value="false"/>
            <property name="reliefPattern" value="falls? *-? *thr(u|ough)"/>
         </module>
    </module>
</module>
```

```
hp@DESKTOP-OPBFIIB MINGW64 ~/Desktop
$ cat test1.java
/*
FallThrough
checkLastCaseGroup = (default)false
reliefPattern = (default)falls?[ -]?thr(u|ough)


*/

package com.puppycrawl.tools.checkstyle.checks.coding.fallthrough;

public class InputFallThrough4
{
    void foo(int i){
        switch(i) {
            case 1:
                 int fallthru = 2; // fallthru
            case 2:// violation 'Fall through from previous branch of the switch statement.'
                 int fallthrough = 2; //            fall thru
            case 3: // violation 'Fall through from previous branch of the switch statement.'
                 break;
            case 4:
                 int fallthru2 = 3;
                 //     fall   -   thru
            case 5:
                 int fallthru3 = 4; // This is a fallthru comment
            case 6: // violation 'Fall through from previous branch of the switch statement.'
                 int fallthru4 = 5; // falldown
            default: // violation 'Fall through from previous branch of the switch statement.'
            }
    }
}
```

```
hp@DESKTOP-OPBFIIB MINGW64 ~/Desktop
$ java -jar checkstyle-10.9.3-SNAPSHOT-all.jar -c config1.xml test1.java
Starting audit...
[ERROR] C:\Users\hp\Desktop\test1.java:17:13: Fall through from previous branch of the switch statement. [FallThrough]
[ERROR] C:\Users\hp\Desktop\test1.java:19:13: Fall through from previous branch of the switch statement. [FallThrough]
[ERROR] C:\Users\hp\Desktop\test1.java:26:13: Fall through from previous branch of the switch statement. [FallThrough]
[ERROR] C:\Users\hp\Desktop\test1.java:28:13: Fall through from previous branch of the switch statement. [FallThrough]
Audit done.
Checkstyle ends with 4 errors.
```

Reply:
Please share also out put on release version of jar. It is not clear what version of code you use, with your fix or without.

There should be no violation on line 26, or we found defect in Check.

Reply:
> hp@DESKTOP-OPBFIIB MINGW64 ~/Desktop
> $ java -jar checkstyle-10.9.3-SNAPSHOT-all.jar -c config1.xml test1.java

@romani I think am using the released version of jar `checkstyle-10.9.3-SNAPSHOT-all.jar`

Reply:
@romani As you said, May be there is defect in check.

Reply:
@coderdeadpool please use release version from https://github.com/checkstyle/checkstyle/releases/tag/checkstyle-10.9.3, and if we can replicate false positive, please create an issue.

Reply:
@nrmancuso @romani I have used the release version and getting the same output as above.

```
$ cat config1.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="FallThrough">
            <property name="checkLastCaseGroup" value="false"/>
            <property name="reliefPattern" value="falls?[ -]?thr(u|ough)"/>
         </module>
    </module>
</module>
```

```
$ cat test1.java
/*
FallThrough
checkLastCaseGroup = (default)false
reliefPattern = (default)falls?[ -]?thr(u|ough)


*/

package com.puppycrawl.tools.checkstyle.checks.coding.fallthrough;

public class InputFallThrough4
{
    void foo(int i){
        switch(i) {
            case 1:
                 int fallthru = 2; // fallthru
            case 2:// violation 'Fall through from previous branch of the switch statement.'
                 int fallthrough = 2; //            fall thru
            case 3: // violation 'Fall through from previous branch of the switch statement.'
                 break;
            case 4:
                 int fallthru2 = 3;
                 //     fall   -   thru
            case 5:
                 int fallthru3 = 4; // This is a fallthru comment
            case 6: // violation 'Fall through from previous branch of the switch statement.'
                 int fallthru4 = 5; // falldown
            default: // violation 'Fall through from previous branch of the switch statement.'
            }
    }
}
```

```
$ java -jar checkstyle-10.9.3-all.jar -c config1.xml test1.java

Starting audit...
[ERROR] C:\Users\hp\Desktop\test1.java:17:13: Fall through from previous branch of the switch statement. [FallThrough]
[ERROR] C:\Users\hp\Desktop\test1.java:19:13: Fall through from previous branch of the switch statement. [FallThrough]
[ERROR] C:\Users\hp\Desktop\test1.java:24:13: Fall through from previous branch of the switch statement. [FallThrough]
[ERROR] C:\Users\hp\Desktop\test1.java:26:13: Fall through from previous branch of the switch statement. [FallThrough]
[ERROR] C:\Users\hp\Desktop\test1.java:28:13: Fall through from previous branch of the switch statement. [FallThrough]
Audit done.
Checkstyle ends with 5 errors.
```
There should be no violation on line no. 26. It might be showing violation due to a defect in the check.

Reply:
@nrmancuso @romani @coderdeadpool  according to me https://github.com/checkstyle/checkstyle/pull/12950#discussion_r1161656762
output
should look like 
```
kevin@inspiron-15-5510:~/Desktop/check_style$ java -jar /home/kevin/Desktop/check_style/checkstyle/checkstyle/target/checkstyle-10.9.4-SNAPSHOT-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/kevin/Desktop/check_style/Test.java:24:13: Fall through from previous branch of the switch statement. [FallThrough]
[ERROR] /home/kevin/Desktop/check_style/Test.java:28:13: Fall through from previous branch of the switch statement. [FallThrough]
Audit done.
Checkstyle ends with 2 errors.
```

Reply:
@Kevin222004 there should be only two valid violations on lines 19 and 28. There is a fall-through from case 2 to case 3(for line 19 violation) and from case 6 to the default case(for line 28 violation) without any comment matching the relief pattern.

Reply:
@Kevin222004 yes there should be only 2 violations. That is on line 24 and 28

Reply:
@coderdeadpool  I agree with 24 and 28 line violation 

but why violation on line 19 case 2 contains `                 int fallthrough = 2; //            fall thru`
where fall thru is expected to match with regex you provide in config

Reply:
@Kevin222004 Yes you are right. But dont know why i am getting these extra violations.

Reply:
don't worry i have updated this check for https://github.com/checkstyle/checkstyle/pull/12966  and tested your code against jar after changes in my local your jar is currently released version mine was after updation of check